### PR TITLE
Message manager clean up

### PR DESCRIPTION
**Pull request by frankiebee**
This is a clean up of message manager 
Clean up message manger includes: 
- Provider egine bump for new message manager flow 
- Remove presence of message manger in keyring controller
- Change the status wording from conf to approved
- make Message manager a class
- fix messages not being apart of the badge
- rewrite message manger to better reflect controller pattern

### CODE REVIEW

File: app/scripts/keyring-controller.js [hunk 1]
```diff
@@ -157,6 +146,17 @@ class KeyringController extends EventEmitter {
...
+  //
+  // returns Promise( @string currentSelectedAccount )
+  //
+  // Removes the current vault's seed words from the UI's state tree,
+  // ensuring they are only ever available in the background process.
+  clearSeedWordCache () {
+    this.configManager.setSeedWords(null)
+    return Promise.resolve(this.configManager.getSelectedAccount())
+  }
+
```
Comment (kumavis) [inline app/scripts/keyring-controller.js]: i think you accidently re-added this

File: app/scripts/keyring-controller.js [hunk 2]
```diff
@@ -207,8 +207,8 @@ class KeyringController extends EventEmitter {
       this.keyrings.push(keyring)
       return this.setupAccounts(accounts)
     })
-    .then(() => this.persistAllKeyrings())
-    .then(() => this.fullUpdate())
+    .then(() => { return this.password })
+    .then(this.persistAllKeyrings.bind(this))
```
Comment (kumavis) [inline app/scripts/keyring-controller.js]: is this change intentional?

File: app/scripts/keyring-controller.js [hunk 3]
```diff
@@ -42,9 +37,7 @@ class KeyringController extends EventEmitter {
     this.ethStore = opts.ethStore
     this.encryptor = encryptor
     this.keyrings = []
-
-    this._unconfMsgCbs = {}
-
+    this.identities = {} // Essentially a name hash
```
Comment (kumavis) [inline app/scripts/keyring-controller.js]: is this an intentional change?

**Review by kumavis — CHANGES_REQUESTED** [review]

File: app/scripts/keyring-controller.js (hunk 3 above)
Comment (frankiebee) [inline app/scripts/keyring-controller.js]: nope ill remove that i think it snuck in during merge conflicts

File: app/scripts/keyring-controller.js (hunk 2 above)
Comment (frankiebee) [inline app/scripts/keyring-controller.js]: lesson learned.. i cant do the conflicts on github

File: app/scripts/lib/message-manager.js [hunk 4]
```diff
@@ -1,61 +1,113 @@
...
+    this._saveMsgList(messages)
   }
-  this._saveMsgList(messages)
-}
 
+  _saveMsgList (msgList) {
+    this.emit('updateBadge')
+    let state = this.memStore.getState()
+    state.messages = msgList
+    this.memStore.putState(state)
```
Comment (kumavis) [inline app/scripts/lib/message-manager.js]:
```js
this.memStore.updateState({ messages: msgList })
```

**Review by kumavis — APPROVED** [review]
